Lib/unittest/mock.py

```diff
@@ -778,8 +778,10 @@ def assert_not_called(_mock_self):
         """
         self = _mock_self
         if self.call_count != 0:
-            msg = ("Expected '%s' to not have been called. Called %s times." %
-                   (self._mock_name or 'mock', self.call_count))
+            msg = ("Expected '%s' to not have been called. Called %s times.\n"
+                   "Calls: %r." % (self._mock_name or 'mock',
```

Comment:
repr(self.mock_calls) can be very long.

unittest.utils has a safe_repr() method which truncates the repr to 80 characters. Maybe you should use it?

Alternative: there is also reprlib.repr(), but I suggest to reuse safe_repr() instead.

Reply:
Agree, scares me out that someone mocks a heavily used functions and this prints all the calls.

Reply:
I agree. But the reason I didn't was that `assert_has_calls` already has this functionality and does not limit the output. I should probably change that as well.